src/megaapi_impl.cpp

```diff
@@ -9825,6 +9830,30 @@ void MegaApiImpl::setMyChatFilesFolder(MegaHandle nodehandle, MegaRequestListene
     delete stringMap;
 }
 
+void MegaApiImpl::getUserAlias(MegaHandle uh, MegaRequestListener *listener)
+{
+    MegaRequestPrivate *request = new MegaRequestPrivate(MegaRequest::TYPE_GET_ATTR_USER, listener);
+    request->setParamType(MegaApi::USER_ATTR_ALIAS);
+    request->setNodeHandle(uh);
+    request->setText(Base64Str<MegaClient::USERHANDLE>(uh));
+    requestQueue.push(request);
+    waiter->notify();
+}
+
+void MegaApiImpl::setUserAlias(MegaHandle uh, const char *alias, MegaRequestListener *listener)
+{
+    MegaRequestPrivate *request = new MegaRequestPrivate(MegaRequest::TYPE_SET_ATTR_USER, listener);
+    MegaStringMap *stringMap = new MegaStringMapPrivate();
```

Comment:
In cases like this where the object will be deleted again a few lines later, it can just be declared on the stack: `MegaStringMapPrivate stringMap;`, and then passed to functions with `&stringMap`, and then there's less heap activity, and no need to manually delete it.    Alternatively the setMegaStringMap could just take ownership, and then there would be no need to make a copy and delete the original, which seems like a lot more work than necessary.  Maybe add a flag on setMegaStringMap as to whether it should assume ownership, or copy?    thanks

Reply:
Thank you, Matt. The `stringMap` can be created in the heap, indeed. I agree. I'll adjust it.
For the setter of the `MegaRequest`, I'd prefer to keep it as it is for historical reasons, where they always do a copy. I'm aware of the extra work, but it helps Java bindings not to mess up with the passed objects. Basically, the SDK never takes ownership of the passed object :)